sklearnex/spmd/linear_model/tests/test_incremental_linear_spmd.py

```diff
@@ -0,0 +1,329 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+from onedal.tests.utils._dataframes_support import (
+    _as_numpy,
+    _convert_to_dataframe,
+    get_dataframes_and_queues,
+)
+from sklearnex.tests._utils_spmd import (
+    _generate_regression_data,
+    _get_local_tensor,
+    _mpi_libs_and_gpu_available,
+)
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.parametrize("macro_block", [None, 1024])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.mpi
+def test_incremental_linear_regression_fit_spmd_gold(
+    dataframe, queue, fit_intercept, macro_block, dtype
+):
+    # Import spmd and non-SPMD algo
+    from sklearnex.linear_model import IncrementalLinearRegression
+    from sklearnex.spmd.linear_model import (
+        IncrementalLinearRegression as IncrementalLinearRegression_SPMD,
+    )
+
+    # Create gold data and process into dpt
+    X = np.array(
+        [
+            [0.0, 0.0],
+            [1.0, 2.0],
+            [2.0, 4.0],
+            [3.0, 8.0],
+            [4.0, 16.0],
+            [5.0, 32.0],
+            [6.0, 64.0],
+            [7.0, 128.0],
+            [8.0, 0.0],
+            [9.0, 2.0],
+            [10.0, 4.0],
+            [11.0, 8.0],
+            [12.0, 16.0],
+            [13.0, 32.0],
+            [14.0, 64.0],
+            [15.0, 128.0],
+        ],
+        dtype=dtype,
+    )
+    dpt_X = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
+    local_X = _get_local_tensor(X)
+    local_dpt_X = _convert_to_dataframe(local_X, sycl_queue=queue, target_df=dataframe)
+
+    y = np.dot(X, [1, 2]) + 3
+    dpt_y = _convert_to_dataframe(y, sycl_queue=queue, target_df=dataframe)
+    local_y = _get_local_tensor(y)
+    local_dpt_y = _convert_to_dataframe(local_y, sycl_queue=queue, target_df=dataframe)
+
+    inclin_spmd = IncrementalLinearRegression_SPMD(fit_intercept=fit_intercept)
+    inclin = IncrementalLinearRegression(fit_intercept=fit_intercept)
+
+    if macro_block is not None:
+        hparams = inclin.get_hyperparameters("fit")
+        hparams.cpu_macro_block = macro_block
+        hparams.gpu_macro_block = macro_block
+
+        hparams_spmd = inclin_spmd.get_hyperparameters("fit")
+        hparams_spmd.cpu_macro_block = macro_block
+        hparams_spmd.gpu_macro_block = macro_block
+
+    inclin_spmd.fit(local_dpt_X, local_dpt_y)
+    inclin.fit(dpt_X, dpt_y)
```

Comment:
Why is just fit() used here?

Reply:
https://github.com/intel/scikit-learn-intelex/pull/1972#discussion_r1745114693